bpf/bpf_lxc.c

```diff
@@ -1979,28 +1972,20 @@ ipv4_policy(struct __ctx_buff *ctx, struct iphdr *ip4, int ifindex, __u32 src_la
 		return verdict;
 
 skip_policy_enforcement:
-#ifdef ENABLE_NODEPORT
+	/* Reset the entry also for CT_REOPENED. This way any difference in source security id,
+	 * proxy redirect, dsr, or nodeport gets set correctly.
+	 */
 	if (ret == CT_NEW || ret == CT_REOPENED) {
-# ifdef ENABLE_DSR
-		/* Clear .dsr_internal flag for old connections: */
-		if (ret == CT_REOPENED && ct_state->dsr_internal)
-			ct_update_dsr(get_ct_map4(tuple), tuple, false);
-# endif /* ENABLE_DSR */
```

Comment:
Let's address this part separately (https://github.com/cilium/cilium/pull/32642), there's more to it.

Reply:
This was conditionally setting the flag to false, removed this as now the flag is false by default. IMO removal of this is the right thing to do in this PR (no functional change for the parts that were already updated for `CT_REOPENED`.